test/ts/unit/collateralized/runners/register_term_start.ts

```diff
@@ -0,0 +1,160 @@
+        let mockDebtRegistry: MockDebtRegistryContract;
+        let mockCollateralToken: MockERC20TokenContract;
+        let mockTokenRegistry: MockTokenRegistryContract;
+
+        let txHash: string;
+
+        describe(scenario.description, () => {
+            before(async () => {
+                ATTACKER = this.accounts.ATTACKER;
+                COLLATERALIZER = this.accounts.COLLATERALIZER;
+                MOCK_DEBT_KERNEL_ADDRESS = this.accounts.MOCK_DEBT_KERNEL_ADDRESS;
+
+                mockCollateralizedTermsContract = this.contracts.mockCollateralizedTermsContract;
+                mockDebtRegistry = this.contracts.mockDebtRegistry;
+                mockCollateralToken = this.contracts.mockCollateralToken;
+                mockTokenRegistry = this.contracts.mockTokenRegistry;
+
+                await mockDebtRegistry.reset.sendTransactionAsync();
+                await mockCollateralToken.reset.sendTransactionAsync();
+                await mockTokenRegistry.reset.sendTransactionAsync();
+
+                // Mock the collateral token's presence in the token registry at
+                // the specified index
+                await mockTokenRegistry.mockGetTokenAddressByIndex.sendTransactionAsync(
+                    scenario.collateralTokenIndexInRegistry,
+                    mockCollateralToken.address,
+                );
+
+                // Mock collateral token's balance for collateralizer
+                await mockCollateralToken.mockBalanceOfFor.sendTransactionAsync(
+                    COLLATERALIZER,
+                    scenario.collateralTokenBalance,
+                );
+
+                // Mock collateral token's allowance for collateralizer
+                await mockCollateralToken.mockAllowanceFor.sendTransactionAsync(
+                    COLLATERALIZER,
+                    mockCollateralizedTermsContract.address,
+                    scenario.collateralTokenAllowance,
+                );
+
+                // Mock `debtRegistry.getTerms` return value
+                await mockDebtRegistry.mockGetTermsReturnValueFor.sendTransactionAsync(
+                    scenario.agreementId,
+                    scenario.termsContract(mockCollateralizedTermsContract.address, ATTACKER),
+                    scenario.termsContractParameters,
+                );
+
+                ABIDecoder.addABI(mockCollateralizedTermsContract.abi);
+            });
+
+            after(() => {
+                ABIDecoder.removeABI(mockCollateralizedTermsContract.abi);
+            });
+
+            if (scenario.succeeds) {
+                it("should register without throwing", async () => {
+                    txHash = await mockCollateralizedTermsContract.registerTermStart.sendTransactionAsync(
+                        scenario.agreementId,
+                        COLLATERALIZER,
+                        { from: scenario.from(MOCK_DEBT_KERNEL_ADDRESS, ATTACKER) },
+                    );
+
+                    expect(true);
```

Comment:
this strikes me as a bit implicit. any way to make this more explicit as to what we're testing?

Reply:
Maybe be something like:

```javascript
expect(async () => {
 await mockCollateralizedTermsContract.registerTermStart.sendTransactionAsync(...))
}).to.not.throw()
```

http://www.chaijs.com/api/bdd/#method_not

Reply:
^^ ideally yes -- I recall running into issues related to getting this to work correctly.  I'll try to see if I can get it to work and fix.

Reply:
Ah ok -- I remember the issue now.  We basically want to assert that it doesn't throw, but we also want to capture the txhash.  Shifting the tests around so that we don't do weird implicit voodoo tests like this.  LMK if it's up to satisfaction.